quickwit-search/src/leaf.rs

```diff
@@ -178,10 +178,19 @@ pub async fn leaf_search(
 
     merged_search_results
         .failed_splits
-        .extend(errors.iter().map(|(split_id, err)| SplitSearchError {
-            split_id: split_id.to_string(),
-            error: format!("{:?}", err),
-            retryable_error: true,
+        .extend(errors.iter().map(|(split_id, err)| {
+            let retryable_error = match err {
+                SearchError::IndexDoesNotExist { index_id: _ } | SearchError::InvalidQuery(_) => {
+                    false
+                }
+                _ => true,
```

Comment:
Can you put the list of items instead of using the default pattern?

Reply:
I moved it to a a method to the enum,  so it should be easier to spot